scripts/base/frameworks/cluster/main.zeek

```diff
@@ -241,6 +254,29 @@ export {
 	## locally whenever a connected cluster node becomes disconnected.
 	global node_down: event(name: string, id: string);
 
+	## When using broker-enabled cluster framework, this event will be sent to
+	## the manager whenever a cluster node has established all peerings to
+	## its fellow cluster nodes based on the given cluster layout.
+	##
+	## .. warning::
+	##
+	##     There is no tracking of cluster node connectivity. Thus, there is
+	##     no guarantee that all peerings still exist at the time of this event
+	##     being raised.
+	global node_connected: event(name: string, id: string);
+
+	## When using broker-enabled cluster framework, this event will be
+	## broadcasted from the manager once all nodes reported that they have
+	## established all peerings to fellow cluster nodes based on the given
+	## cluster layout.
+	##
+	## .. warning::
+	##
+	##     There is no tracking of cluster node connectivity. Thus, there is
+	##     no guarantee that all peerings still exist at the time of this event
+	##     being raised.
+	global cluster_init: event();
```

Comment:
`cluster_init()` is very generic and possibly "over-promising": One proposal would to call it: `cluster_connected()` with the same semantics as in this PR (all nodes connected to their peers and broadcasted).

As for the "over-promising", there are other cluster related functionality that may not yet be done, but just seeing `cluster_init()` may well be considered completed, too:
* Intel file distribution
* Config framework distribution
* Anything in third-party scripts related to clusters

Could you see `cluster_connected()`, too?

I'll look at the rest later. Had chatted with @rsmmr about this there's definitely a need for such an event as it has been requested enough in non-testing contexts before.

Reply:
Actually I started with the pair of `cluster_connected` and `cluster_disconnected`. However, the latter would have added some more complexity. As I couldn't come up with a pressing use case for `cluster_disconnected` I dropped it and switched from "connected" to "init". So I am happy to go back to `cluster_connected`.

Reply:
Sounds good. `cluster_connected` should only happen once and be documented as such, that's easy to understand and implement "correctly". Otherwise a restarting node may confuse script logic that should've been simple an assumed it only happens once.

If a node crashes comes back again, can have a separate `cluster_reconnected` event for that in the future if it's needed (or can also impelement node_down's by hand.

(all Robin's suggestion - none mine :-) )

I'd forego the `cluster_disconnected` event for now.

Reply:
> `cluster_connected` should only happen once and be documented as such, that's easy to understand and implement "correctly". Otherwise a restarting node may confuse script logic that should've been simple an assumed it only happens once.

I am not sure I follow this reasoning. I would have assumed that any non-test code which cares about a global *cluster ready* event would also care about the cluster not being ready and take some action on that. Leaving users in the illusion that cluster state is static after startup might actually cause more confusion in the end, not less.

My gut feeling would have been that we'd want to explicitly document how this event can be meaningfully used and what caveats it has (ideally with workarounds for e.g., nodes coming and going).